Example warning:

```
/tmp/tmpyykg1br1/parse.c:32:35: warning: passing argument 2 of ‘_PyPegen_lookahead’ from incompatible pointer type [-Wincompatible-pointer-types]
   32 |             _PyPegen_lookahead(1, _PyPegen_name_token, p)
      |                                   ^~~~~~~~~~~~~~~~~~~
      |                                   |
      |                                   struct _expr * (*)(Parser *) {aka struct _expr * (*)(struct <anonymous> *)}
In file included from /tmp/tmpyykg1br1/parse.c:2:
/home/pablogsal/github/python/master/Parser/pegen/pegen.h:90:36: note: expected ‘void * (*)(Parser *)’ {aka ‘void * (*)(struct <anonymous> *)’} but argument is of type ‘struct _expr * (*)(Parser *)’ {aka ‘struct _expr * (*)(struct <anonymous> *)’}
   90 | int _PyPegen_lookahead(int, void *(func)(Parser *), Parser *);
      |                             ~~~~~~~^~~~~~~~~~~~~~~
ok
test_mutually_left_recursive (test.test_peg_generator.test_c_parser.TestCParser) ... ok
test_nasty_mutually_left_recursive (test.test_peg_generator.test_c_parser.TestCParser) ... ok
test_negative_lookahead (test.test_peg_generator.test_c_parser.TestCParser) ... /tmp/tmp0fvwxl48/parse.c: In function ‘start_rule’:
/tmp/tmp0fvwxl48/parse.c:32:35: warning: passing argument 2 of ‘_PyPegen_lookahead’ from incompatible pointer type [-Wincompatible-pointer-types]
   32 |             _PyPegen_lookahead(0, _PyPegen_name_token, p)
      |                                   ^~~~~~~~~~~~~~~~~~~
      |                                   |
      |                                   struct _expr * (*)(Parser *) {aka struct _expr * (*)(struct <anonymous> *)}
In file included from /tmp/tmp0fvwxl48/parse.c:2:
/home/pablogsal/github/python/master/Parser/pegen/pegen.h:90:36: note: expected ‘void * (*)(Parser *)’ {aka ‘void * (*)(struct <anonymous> *)’} but argument is of type ‘struct _expr * (*)(Parser *)’ {aka ‘struct _expr * (*)(struct <anonymous> *)’}
   90 | int _PyPegen_lookahead(int, void *(func)(Parser *), Parser *);

```

--- a/Parser/pegen/pegen.c
+++ b/Parser/pegen/pegen.c
@@ -640,6 +640,16 @@ _PyPegen_is_memoized(Parser *p, int type, void *pres)
     return 0;
 }
 
+
+int
+_PyPegen_lookahead_with_name(int positive, expr_ty (func)(Parser *), Parser *p)
+{
+    int mark = p->mark;
+    void *res = func(p);
+    p->mark = mark;
+    return (res != NULL) == positive;
+}
+
 int
 _PyPegen_lookahead_with_string(int positive, void *(func)(Parser *, const char *), Parser *p,
                       const char *arg)
@@ -663,7 +673,7 @@ int
 _PyPegen_lookahead(int positive, void *(func)(Parser *), Parser *p)
 {
     int mark = p->mark;
-    void *res = func(p);
+    void *res = (void*)func(p);
     p->mark = mark;
     return (res != NULL) == positive;
 }

--- a/Parser/pegen/pegen.h
+++ b/Parser/pegen/pegen.h
@@ -85,6 +85,7 @@ int _PyPegen_insert_memo(Parser *p, int mark, int type, void *node);
 int _PyPegen_update_memo(Parser *p, int mark, int type, void *node);
 int _PyPegen_is_memoized(Parser *p, int type, void *pres);
 
+int _PyPegen_lookahead_with_name(int, expr_ty (func)(Parser *), Parser *);
 int _PyPegen_lookahead_with_string(int, void *(func)(Parser *, const char *), Parser *, const char *);
 int _PyPegen_lookahead_with_int(int, Token *(func)(Parser *, int), Parser *, int);
 int _PyPegen_lookahead(int, void *(func)(Parser *), Parser *);

--- a/Tools/peg_generator/pegen/build.py
+++ b/Tools/peg_generator/pegen/build.py
@@ -8,6 +8,7 @@ import distutils.log
 from distutils.core import Distribution, Extension
 from distutils.command.clean import clean  # type: ignore
 from distutils.command.build_ext import build_ext  # type: ignore
+from distutils.tests.support import fixup_build_ext
 
 from pegen.c_generator import CParserGenerator
 from pegen.grammar import Grammar
@@ -66,6 +67,7 @@ def compile_c_extension(
     ]
     dist = Distribution({"name": extension_name, "ext_modules": extension})
     cmd = build_ext(dist)
+    fixup_build_ext(cmd)
     cmd.inplace = True
     if build_dir:
         cmd.build_temp = build_dir

--- a/Tools/peg_generator/pegen/c_generator.py
+++ b/Tools/peg_generator/pegen/c_generator.py
@@ -93,7 +93,9 @@ class CCallMakerVisitor(GrammarVisitor):
         func, args = call.split("(", 1)
         assert args[-1] == ")"
         args = args[:-1]
-        if not args.startswith("p,"):
+        if "name_token" in call:
+            return None, f"_PyPegen_lookahead_with_name({positive}, {func}, {args})"
+        elif not args.startswith("p,"):
             return None, f"_PyPegen_lookahead({positive}, {func}, {args})"
         elif args[2:].strip().isalnum():
             return None, f"_PyPegen_lookahead_with_int({positive}, {func}, {args})"

